src/logic/addressBook/store/reducer/addressBook.js

```diff
@@ -1,19 +1,40 @@
 // @flow
 import { Map } from 'immutable'
 import { handleActions, type ActionType } from 'redux-actions'
-import type { Cookie } from '~/logic/cookies/model/cookie'
+import type { AddressBookEntry } from '~/logic/addressBook/model/addressBook'
+import { ADD_ENTRY } from '~/logic/addressBook/store/actions/addAddressBookEntry'
+import { UPDATE_ENTRY } from '~/logic/addressBook/store/actions/updateAddressBookEntry'
+import { REMOVE_ENTRY } from '~/logic/addressBook/store/actions/removeAddressBookEntry'
 import { ADD_ADDRESS_BOOK } from '~/logic/addressBook/store/actions/addAddressBook'
 
 export const ADDRESS_BOOK_REDUCER_ID = 'addressBook'
 
-export type State = Map<string, Map<string, Cookie>>
+export type State = Map<string, Map<string, AddressBookEntry>>
 
 export default handleActions<State, *>(
   {
     [ADD_ADDRESS_BOOK]: (state: State, action: ActionType<Function>): State => {
-      const { addressBook } = action.payload
+      const { addressBook, safeAddress } = action.payload
+      return state.setIn(['addressBook', safeAddress], addressBook)
+    },
+    [ADD_ENTRY]: (state: State, action: ActionType<Function>): State => {
+      const { entry, safeAddress } = action.payload
 
-      return state.set('addressBook', addressBook)
+      const currentList = state.getIn(['addressBook', safeAddress])
+      currentList.push(entry)
+      return state.setIn(['addressBook', safeAddress], currentList)
+    },
+    [UPDATE_ENTRY]: (state: State, action: ActionType<Function>): State => {
+      const { entry, entryIndex, safeAddress } = action.payload
+      const entriesList = state.getIn(['addressBook', safeAddress])
+      entriesList[entryIndex] = entry
```

Comment:
is this a list or an array? if this is list, why not to use methods provided by immutable.js? Also I have flow error about `.setIn` being undefined in this file, could you please check if this is fixable?

Reply:
It's an array

Reply:
why did you decide to use array instead of immutable.js list?

Reply:
It's a lot easier to maintain, I can change it, but it's quite simpler using just an array

Reply:
What are the difficulties in maintaining a list? They have the same methods except that you need to use `.set`

Reply:
Done